tests/test_server.py

```diff
@@ -686,6 +686,40 @@ def test_training_payload_from_yaml_save_to_default_model_directory(
     assert payload.get("output") == expected
 
 
+@pytest.mark.parametrize(
+    "headers, expected",
+    [
+        ({}, {"augmentation_factor": 50}),
+        ({"augmentation": "25"}, {"augmentation_factor": 25}),
+    ],
+)
+def test_training_payload_from_yaml_core_arguments(
```

Comment:
I think it would be more robust to test with an actual http request (e.g. like done in [this test function](https://github.com/RasaHQ/rasa/blob/a1efa3fa854d88460d9e7b0a0a4a6dbbfefeb19a/tests/test_server.py#L592)). This has the advantange that the test focuses on the functionality and less on the specific implementation (`_training_payload_from_yaml`)

Reply:
If you use `monkeypatch` to mock the actual training, you can test whether the parameters were passed correctly.

Reply:
@wochinge sure, I've added a new test at https://github.com/RasaHQ/rasa/pull/8132/files#diff-5620588ecb93aec064fe16ec7e5c10f6669041c927717e0f57b3f9a44ffec5acR641

Reply:
awesome! I think we can even delete this and the ones after this then.